src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/ReviewUnusedParameters.Fixer.cs

```diff
@@ -22,9 +30,133 @@ public sealed override FixAllProvider GetFixAllProvider()
+            // TODO: throw a weak exception to get diagnostic of the situaiton.
+            return null;
+        }
+
+        private async Task<ImmutableArray<KeyValuePair<DocumentId, SyntaxNode>>> GetNodesToRemoveAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            SyntaxNode root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            SyntaxNode node = root.FindNode(diagnostic.Location.SourceSpan);
+            node = GetParameterNode(node);
+
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            ISymbol parameterSymbol = editor.SemanticModel.GetDeclaredSymbol(node);
+            ISymbol methodDeclarationSymbol = editor.SemanticModel.GetDeclaredSymbol(node.Parent.Parent);
+            if (methodDeclarationSymbol is IPropertySymbol)
+            {
+                // Should not fix removing unused property indexer. Cancel the fix.
+                return ImmutableArray<KeyValuePair<DocumentId, SyntaxNode>>.Empty;
+            }
+
+            var nodesToRemove = ImmutableArray.CreateBuilder<KeyValuePair<DocumentId, SyntaxNode>>();
+            nodesToRemove.Add(new KeyValuePair<DocumentId, SyntaxNode>(document.Id, node));
+            var referencedSymbols = await SymbolFinder.FindReferencesAsync(methodDeclarationSymbol, document.Project.Solution, cancellationToken).ConfigureAwait(false);
+
+            foreach (var referencedSymbol in referencedSymbols)
+            {
+                if (referencedSymbol.Locations != null)
+                {
+                    foreach (var referenceLocation in referencedSymbol.Locations)
+                    {
+                        Location location = referenceLocation.Location;
+                        var referenceRoot = location.SourceTree.GetRoot();
+                        var referencedSymbolNode = referenceRoot.FindNode(location.SourceSpan);
+                        DocumentEditor localEditor = await DocumentEditor.CreateAsync(referenceLocation.Document, cancellationToken).ConfigureAwait(false);
+                        var operation = localEditor.SemanticModel.GetOperation(referencedSymbolNode, cancellationToken);
+                        var arguments = GetOperationArguments(referencedSymbolNode, localEditor.SemanticModel, cancellationToken);
+
+                        if (arguments != null)
+                        {
+                            foreach (IArgumentOperation argument in arguments)
+                            {
+                                if (argument.Parameter.Equals(parameterSymbol) && (argument.ArgumentKind == ArgumentKind.Explicit))
+                                {
+                                    if (argument.Value is ILiteralOperation || argument.Value is IParameterSymbol || argument.Value is IArrayCreationOperation || argument.Value is ILocalReferenceOperation)
```

Comment:
Instead of `argument.Value is ILiteralOperation`, can we just check if `argument.Value.ConstantValue.HasValue`? It will handle cases where you have non-literal constant expressions, such as say `1+1`.

Reply:
> argument.Value is IParameterSymbol

This should be `IParameterReferenceOperation`. Also it is preferable to have a switch statement based on `argument.Value.OperationKind` instead of multiple `is` expressions

Reply:
> argument.Value is IArrayCreationOperation

What is array creation has an element initialized with method call? Not sure if it is safe to remove this.

Reply:
What about `IFieldReferenceOperation`? That would be a very common case as field values are passed as arguments, but we need to be careful to remove it. For cases where we have implicit receiver (direct use of `field` with implicit 'this'), explicit this receiver (`this.field`) or null receiver (static field access) or receiver which is one of the other allowed kinds (operation kinds such as parameter reference, local reference or field reference, e.g. `param.field`, `local.field` or `param.Field1.Field2`), we can remove it, but otherwise we cannot (say `Property1.Field`, `MethodCall().Field`, etc) . You can use `IFieldReferenceOperation.Instance` to identify all these cases.

Reply:
Actually, for any cases where `IFieldReferenceOperation.Instance` is non-null, but not an implicit/explicit `this` receiver, the member access can cause a null ref in original code, so we cannot remove it.

Reply:
We may also want to remove `IConversionOperation`, if it is an implicit non-user defined conversion and it's operand is one of the allowed cases.

Reply:
We can still consider handling the most basic cases: IFieldReferenceOperation with (1) null Instance (2) Instance with IsImplicit = true, (3) Instance with OperationKind.InstanceReference